src/test/java/io/strimzi/kafka/bridge/clients/Producer.java

```diff
@@ -148,4 +132,44 @@ private Properties fillDefaultProperties() {
     public Properties getProperties() {
         return properties;
     }
+
+    public static class Builder {
+        private final CompletableFuture<Integer> resultPromise;
+        private final IntPredicate msgCntPredicate;
+        private final String topic;
+        private final String message;
+        private final int partition;
+
+        private Properties properties;
+        private List<KafkaHeader> headers = Collections.emptyList();
+        private boolean withNullKeyRecord = false;
+
+        public Builder(CompletableFuture<Integer> resultPromise, IntPredicate msgCntPredicate, String topic,
+                       String message, int partition) {
+            this.resultPromise = resultPromise;
+            this.msgCntPredicate = msgCntPredicate;
+            this.topic = topic;
+            this.message = message;
+            this.partition = partition;
+        }
+
+        public Builder properties(Properties properties) {
+            this.properties = properties;
+            return this;
+        }
+
+        public Builder headers(List<KafkaHeader> headers) {
+            this.headers = headers;
+            return this;
+        }
```

Comment:
should `headers` and `properties` prefixed with `with` (as for the null key record)?